### PR TITLE
Add Builder::clang_args (plural)

### PR DESCRIPTION
**Pull request by SimonSapin**
Use `AsRef<str>` rather than `Into<String>` because `&&str` (what you get when iterating `&[&str]`) does not implement the latter.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -555,6 +555,15 @@ impl Builder {
         self
     }
 
+    /// Add arguments to be passed straight through to clang.
+    pub fn clang_args<I>(mut self, iter: I) -> Builder
+    where I: IntoIterator, I::Item: AsRef<str> {
```
Comment (emilio) [inline src/lib.rs]: Can we use the format specified in `rustfmt.toml`?

That is:

```rust
pub fn clang_args<I>(mut self, iter: I) -> Builder
    where I: IntoIterator,
          I::Item: AsRef<str>,
{
    // ...
}

File: src/lib.rs [hunk 2]
```diff
@@ -555,6 +555,15 @@ impl Builder {
...
+        for arg in iter {
+            self.options.clang_args.push(arg.as_ref().into());
```
Comment (emilio) [inline src/lib.rs]: perhaps:

```rust
self = self.clang_arg(arg.as_ref());
```

(not sure if you can assing to self, but perhaps using an intermediate variable is an option?)

That way if we add more stuff to `clang_arg` we can just keep this. Not a big deal if it's annoying for some reason.

**Comment by emilio** [comment]
Oh, also, you need to update the cargo.lock so travis is happy.

**Comment by SimonSapin** [comment]
Done both. I was also skeptical about assigning to `self`, but that turns out ok.

@bors-servo r=emilio

**Pull request merged**